### PR TITLE
Feature/update to polkadot v0 9 19

### PR DESCRIPTION
**Pull request by MRamanenkau**

### CODE REVIEW

File: bin/node/runtime/src/lib.rs [hunk 1]
```diff
@@ -867,24 +864,40 @@ impl pallet_treasury::Config for Runtime {
 	type MaxApprovals = MaxApprovals;
 }
 
+parameter_types! {
+	pub const BountyCuratorDeposit: Permill = Permill::from_percent(50);
+	pub const BountyValueMinimum: Balance = 10 * DOLLARS;
+	pub const BountyDepositBase: Balance = 1 * DOLLARS;
+	pub const CuratorDepositMultiplier: Permill = Permill::from_percent(50);
+	pub const CuratorDepositMin: Balance = 1 * DOLLARS;
+	pub const CuratorDepositMax: Balance = 100 * DOLLARS;
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs:873]: Here is the [doc](https://docs.rs/pallet-bounties/latest/pallet_bounties/) but it's not clear what is "body of work - or specified set of objectives".
Polkadot values:
```
pub const BountyDepositBase: Balance = 1 * DOLLARS;
pub const BountyDepositPayoutDelay: BlockNumber = 8 * DAYS;
pub const BountyUpdatePeriod: BlockNumber = 90 * DAYS;
pub const MaximumReasonLength: u32 = 16384;
pub const CuratorDepositMultiplier: Permill = Permill::from_percent(50);
pub const CuratorDepositMin: Balance = 10 * DOLLARS;
pub const CuratorDepositMax: Balance = 200 * DOLLARS;
pub const BountyValueMinimum: Balance = 10 * DOLLARS;
```

File: bin/node/runtime/src/lib.rs [hunk 2]
```diff
@@ -645,21 +656,14 @@ impl pallet_election_provider_multi_phase::Config for Runtime {
...
+	type Solver = SequentialPhragmen<AccountId, SolutionAccuracyOf<Self>, OffchainRandomBalancing>;
 	type ForceOrigin = EnsureRootOrHalfCouncil;
+	type MaxElectableTargets = ConstU16<{ u16::MAX }>;
+	type MaxElectingVoters = MaxElectingVoters;
 	type BenchmarkingConfig = ElectionProviderBenchmarkConfig;
-	// BagsList allows a practically unbounded count of nominators to participate in NPoS elections.
-	// To ensure we respect memory limits when using the BagsList this must be set to a number of
-	// voters we know can fit into a single vec allocation.
-	type VoterSnapshotPerBlock = ConstU32<10_000>;
+	type WeightInfo = pallet_election_provider_multi_phase::weights::SubstrateWeight<Self>;
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs:666]: I'm reviewing this changes

File: bin/node/runtime/src/lib.rs [hunk 3]
```diff
@@ -1668,7 +1682,8 @@ mod tests {
 
 	#[test]
 	fn perbill_as_onchain_accuracy() {
-		type OnChainAccuracy = <Runtime as onchain::Config>::Accuracy;
+		type OnChainAccuracy =
+			<<Runtime as pallet_election_provider_multi_phase::Config>::Solution as NposSolution>::Accuracy;
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs:1686]: We need to review this set of changes

File: bin/node/runtime/src/lib.rs [hunk 4]
```diff
@@ -645,21 +656,14 @@ impl pallet_election_provider_multi_phase::Config for Runtime {
...
-	type Fallback = pallet_election_provider_multi_phase::NoFallback<Self>;
-	type GovernanceFallback =
-		frame_election_provider_support::onchain::OnChainSequentialPhragmen<Self>;
-	type Solver = frame_election_provider_support::SequentialPhragmen<
-		AccountId,
-		pallet_election_provider_multi_phase::SolutionAccuracyOf<Self>,
-		OffchainRandomBalancing,
-	>;
-	type WeightInfo = pallet_election_provider_multi_phase::weights::SubstrateWeight<Self>;
+	type Fallback = onchain::BoundedExecution<OnChainSeqPhragmen>;
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs:659]: Added a [fallback](https://github.com/paritytech/substrate/pull/10988) for failed election

File: bin/node/runtime/src/lib.rs [hunk 5]
```diff
@@ -645,21 +656,14 @@ impl pallet_election_provider_multi_phase::Config for Runtime {
...
+	type GovernanceFallback = onchain::BoundedExecution<OnChainSeqPhragmen>;
+	type Solver = SequentialPhragmen<AccountId, SolutionAccuracyOf<Self>, OffchainRandomBalancing>;
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs:661]: [Added a fallback on failed elections](https://github.com/paritytech/substrate/pull/10988)

File: bin/node/runtime/src/lib.rs [hunk 6]
```diff
@@ -645,21 +656,14 @@ impl pallet_election_provider_multi_phase::Config for Runtime {
...
 	type ForceOrigin = EnsureRootOrHalfCouncil;
+	type MaxElectableTargets = ConstU16<{ u16::MAX }>;
+	type MaxElectingVoters = MaxElectingVoters;
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs:664]: MaxElectingVoters is 10000 in our runtime and [22500 in Polkadot](https://github.com/paritytech/polkadot/blob/master/runtime/polkadot/src/lib.rs#L404).  MaxElectableTargets are the same

File: bin/node/runtime/src/lib.rs [hunk 7]
```diff
@@ -619,9 +620,19 @@ impl frame_support::pallet_prelude::Get<Option<(usize, sp_npos_elections::Extend
...
+	type Solver = SequentialPhragmen<
+		AccountId,
+		pallet_election_provider_multi_phase::SolutionAccuracyOf<Runtime>,
+	>;
+	type DataProvider = <Runtime as pallet_election_provider_multi_phase::Config>::DataProvider;
+}
+
+impl onchain::BoundedExecutionConfig for OnChainSeqPhragmen {
+	type VotersBound = ConstU32<20_000>;
+	type TargetsBound = ConstU32<2_000>;
```
Comment (MRamanenkau) [inline bin/node/runtime/src/lib.rs:635]: It's [fallback](https://github.com/paritytech/substrate/pull/10988) mechanism config

File: bin/node-template/node/src/command_helper.rs [hunk 8]
```diff
@@ -0,0 +1,131 @@
+// This file is part of Substrate.
```
Comment (AndreiNavoichyk) [inline bin/node-template/node/src/command_helper.rs]: Please, remove

**Review by AndreiNavoichyk — DISMISSED** [review]

File: bin/node-template/node/src/command_helper.rs (hunk 8 above)
Comment (MRamanenkau) [reply bin/node-template/node/src/command_helper.rs]: Removed

**Review by AndreiNavoichyk — APPROVED** [review]

**Pull request merged**